### PR TITLE
VIVI-1374 Throw error if no URL, return default for others

### PR DESCRIPTION
**Pull request by deanylev**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,12 +13,17 @@ module.exports.process = (output, origin) => {
     subtitleUrl = `${origin}/ytdl/vtt?suburi=${encodeURIComponent(subtitleFile.subs.url)}`;
   }
 
+  const { url } = data;
```
Comment (DuBistKomisch) [inline index.js]: should this part be immediately after the JSON.parse?

File: index.js [hunk 2]
```diff
@@ -13,12 +13,17 @@ module.exports.process = (output, origin) => {
...
+  if (!url) {
+    throw 'no url';
+  }
+
   return {
-    title: data.title,
-    url: data.url,
+    title: data.title || 'Unknown',
+    url,
     duration: data.duration,
```
Comment (DuBistKomisch) [inline index.js]: should this be `|| 0`?

Comment (deanylev) [reply index.js]: Yeah I was thinking that, can't hurt

File: index.js (hunk 1 above)
Comment (deanylev) [reply index.js]: Sure

**Review by vietbui — APPROVED** [review]

**Review by DuBistKomisch — APPROVED** [review]

**Pull request merged**